src/ui/public/agg_types/buckets/create_filter/date_terms.js

```diff
@@ -0,0 +1,39 @@
+import moment from 'moment';
+import mondayFirstWeekdays from 'ui/utils/monday_first_weekdays';
+
+export default function createTermsFilterProvider(Private) {
+  return function (aggConfig, key) {
+    const dateMethod = aggConfig.params.date_method;
+    const field = aggConfig.params.field;
+    const indexPattern = aggConfig.vis.indexPattern;
+    const weekdays = mondayFirstWeekdays();
+
+    //convert date strings back into date indexes
+    let value = key;
+    switch (aggConfig.params.date_method) {
+      case 'monthOfYear':
+        value = moment.monthsShort().indexOf(key) + 1;
+        break;
+      case 'dayOfWeek':
+        value = weekdays.indexOf(key) + 1;
+        break;
+    }
+
+    return {
+      meta: {
+        index: indexPattern.id,
+        field: field.name,
+        formattedValue: dateMethod + ': ' + key
+      },
+      script: {
+        script: {
+          inline: '(doc[\'' + field.name + '\'].date.' + dateMethod + ') == value',
```

Comment:
Please [use template strings to interpolate strings](https://github.com/elastic/kibana/blob/master/style_guides/js_style_guide.md#use-template-strings-to-interpolate-variables-into-strings)